### PR TITLE
reviving interpreter

### PR DESCRIPTION
**Pull request by lewurm**
only executes a couple regression tests so far

### CODE REVIEW

**Comment by migueldeicaza** [comment]
We probably can leave the ChangeLog file out and the man page, as it is out of date anyawys.

**Comment by vargaz** [comment]
The interpreter should be a part of the normal runtime instead of using its own startup code, etc.

**Comment by kumpera** [comment]
We should not bring the man page and ChangeLog.

This is a good starting point. The next step is to kill mono_interp_init and merge all the installed callbacks to handle both compiled and interpreted code.

**Comment by lewurm** [comment]
Hey @kumpera and @vargaz. The init stuff is still not properly done, but everything is hidden by a configure flag.  It would be nice to get this into `master` and then get some basic CI going similar to what we have for COOP. I already added something to the CI script https://github.com/mono/mono/pull/3563/commits/ffd706c7f68a6722eebd88b2107bd8fe70e9460d , the Jenkins lane is still missing for it though. The configure flag to enable: `--with-interpreter=yes`. And the `make` target for running the enabled regression tests is `make richeck` in `./mono/mini`.

**Comment by lewurm** [comment]
@kumpera updated

**Comment by kumpera** [comment]
I think the doc on the trampoline could better explain how many args it takes and what's done with them.

File: mono/mini/tramp-amd64.c [hunk 1]
```diff
@@ -969,6 +969,88 @@ mono_arch_create_sdb_trampoline (gboolean single_step, MonoTrampInfo **info, gbo
...
+	guint8 *start = NULL, *code, *exits[4], *leave_tramp;
+	MonoJumpInfo *ji = NULL;
+	GSList *unwind_ops = NULL;
+	static int arg_regs[] = {AMD64_ARG_REG1, AMD64_ARG_REG2, AMD64_ARG_REG3, AMD64_ARG_REG4};
+	int i;
+
+	start = code = (guint8 *) mono_global_codeman_reserve (256);
+
+	/* save MethodArguments* onto stack */
+	amd64_push_reg (code, AMD64_ARG_REG2);
```
Comment (kumpera) [inline mono/mini/tramp-amd64.c]: This and other stack manipulation require unwind info to be generated.

Zoltan can explain how to add it.

**Pull request merged**